File: calc.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        int somma = 0;
        int differenza = 0;
        int prodotto = 0;
        int quoziente = 0;
        int resto = 0;
        int potenza = 0;
        int a = 0;
        int b = 0;
        char confermauscita = ' ';
        char invio = ' ';

        Console.WriteLine("SUPER CALCOLATRICE DI CALOGERO IN C#");
        Console.WriteLine("Premi un tasto per continuare, oppure premi 'v' per stampare le info di questo eseguibile");
        invio = Console.ReadKey().KeyChar;
        
        if (invio == 'v')
        {
            Console.WriteLine("Versione 1.0");
            return;
        }
        else
        {
            Console.WriteLine("Inserisci il primo numero");
            a = Convert.ToInt32(Console.ReadLine());
            Console.WriteLine("Inserisci il secondo numero");
            b = Convert.ToInt32(Console.ReadLine());
        }

        if (b == 0)
        {
            somma = a;
            differenza = a;
            prodotto = 0;
            Console.WriteLine("La somma è: " + somma);
            Console.WriteLine("La differenza è: " + differenza);
            Console.WriteLine("Il prodotto è: " + prodotto);
            Console.WriteLine("La divisione è impossibile");
        }
        else
        {
            somma = a + b;
            differenza = a - b;
            prodotto = a * b;
            quoziente = a / b;
            resto = a % b;
            potenza = (int)Math.Pow(a, b);

            Console.WriteLine("La somma è: " + somma);
            Console.WriteLine("La differenza è: " + differenza);
            Console.WriteLine("Il prodotto è: " + prodotto);
            Console.WriteLine("Il quoziente è: " + quoziente);
            Console.WriteLine("Il resto della divisione è: " + resto);
            Console.WriteLine("Il risultato della potenza è: " + potenza);

            Console.WriteLine("Uscire? (y/n):");
            confermauscita = Console.ReadKey().KeyChar;
            if (confermauscita == 'y' || confermauscita == 'Y')
            {
                return;
            }
            Console.WriteLine("Digitare ./calc per riavviare il programma");
        }
    }
}

File: calc.swift
import Foundation

func main() {
    var somma = 0
    var differenza = 0
    var prodotto = 0
    var quoziente = 0
    var resto = 0
    var potenza = 0
    var a = 0
    var b = 0
    var confermaUscita: Character = " "
    var invio: Character = " "
    
    print("SUPER CALCOLATRICE DI CALOGERO IN SWIFT")
    print("Premi un tasto per continuare, oppure premi 'v' per stampare le info di questo eseguibile")
    invio = Character(readLine() ?? "")
    
    if invio == "v" {
        print("Versione 1.0")
        return
    } else {
        print("Inserisci il primo numero")
        a = Int(readLine() ?? "") ?? 0
        print("Inserisci il secondo numero")
        b = Int(readLine() ?? "") ?? 0
    }
    
    if b == 0 {
        somma = a
        differenza = a
        prodotto = 0
        print("La somma è: \(somma)")
        print("La differenza è: \(differenza)")
        print("Il prodotto è: \(prodotto)")
        print("La divisione è impossibile")
    } else {
        somma = a + b
        differenza = a - b
        prodotto = a * b
        quoziente = a / b
        resto = a % b
        potenza = Int(pow(Double(a), Double(b)))
        
        print("La somma è: \(somma)")
        print("La differenza è: \(differenza)")
        print("Il prodotto è: \(prodotto)")
        print("Il quoziente è: \(quoziente)")
        print("Il resto della divisione è: \(resto)")
        print("Il risultato della potenza è: \(potenza)")
        
        print("Uscire? (y/n):")
        confermaUscita = Character(readLine() ?? "")
        if confermaUscita == "y" || confermaUscita == "Y" {
            return
        }
        print("Digitare ./calc per riavviare il programma")
    }
}

main()
